src/splitRegistry.js

```diff
@@ -1,41 +1,38 @@
-var SplitRegistry = function(splitArray) {
-  this._splitArray = splitArray;
-  this._loaded = splitArray !== null;
-};
-
-SplitRegistry.prototype.getSplit = function(splitName) {
-  return this.getSplits()[splitName];
-};
-
-SplitRegistry.prototype.isLoaded = function() {
-  return this._loaded;
-};
-
-SplitRegistry.prototype.asV1Hash = function() {
-  var v1Hash = {};
-  for (var splitName in this.getSplits()) {
-    var split = this._splits[splitName];
-    v1Hash[splitName] = split.getWeighting();
+class SplitRegistry {
+  constructor(splitArray) {
+    this._splitArray = splitArray;
+    this._loaded = splitArray !== null;
   }
 
-  return v1Hash;
-};
+  getSplit(splitName) {
+    return this.getSplits()[splitName];
+  }
 
-SplitRegistry.prototype.getSplits = function() {
-  if (!this._loaded) {
-    return {};
+  isLoaded() {
+    return this._loaded;
   }
 
-  if (!this._splits) {
-    this._splits = {};
-    this._splitArray.forEach(
-      function(split) {
-        this._splits[split.getName()] = split;
-      }.bind(this)
-    );
+  asV1Hash() {
+    let v1Hash = {};
```

Comment:
minor nit, could this be `const`?

Reply:
Oh yes, I forgot Object keys are not protected by `const`. Will update.